### PR TITLE
[MOV] base: pure RPC endpoints to their own module

### PR DESCRIPTION
**Pull request by xmo-odoo**

### CODE REVIEW

**Review by Julien00859 — COMMENTED** [review]
Thanks for the bits inside test_http.

If you're interested in making each change inside its dedicated commit, there are bits about server-wide-module inside the first commit that I think should be in a later commit.

Shall we enforce `rpc` in the server-wide-modules? e.g. should `rpc` be present if one does `--load foo`? https://github.com/odoo/odoo/blob/90c03e0d394cf91c066a496af2508aa12a75c10a/odoo/service/server.py#L1248-L1249

**Review by Julien00859 — APPROVED** [review]

File: odoo/addons/test_rpc/__manifest__.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
     "version": "1.0",
     "category": "Hidden/Tests",
     "description": """A module to test the RPC requests.""",
-    "depends": ["base", "web"],
+    "depends": ["web", "rpc"],
```
Comment (kmagusiak) [inline odoo/addons/test_rpc/__manifest__.py:7]: Is `web` still needed here? I suppose there are some testing tools required.

**Review by kmagusiak — COMMENTED** [review]
Maybe `load_server_wide_modules()` needs to be adapted too. We forcefully load base and web, but maybe web could be removed from there.

**Comment by Julien00859** [comment]
> Maybe `load_server_wide_modules()` needs to be adapted too. We forcefully load base and web, but maybe web could be removed from there.

This would mean that /web/database/selector wouldn't be accessible on multi-tenant servers with a bogus `--load`

**Review by kmagusiak — APPROVED** [review]

**Review by rco-odoo — APPROVED** [review]
@robodoo rebase-ff r+

**Pull request closed without merging**